aten/src/ATen/core/dispatch/OpSchema.h

```diff
@@ -1,13 +1,18 @@
 #pragma once
 
 #include <ATen/core/dispatch/DispatchKey.h>
+#include <ATen/core/ivalue.h>
 #include <c10/util/Array.h>
 #include <c10/util/Metaprogramming.h>
+#include <c10/util/TypeList.h>
 #include <c10/core/DeviceType.h>
-#include <c10/core/Tensor.h>
+#include <ATen/core/Tensor.h>
 
 namespace c10 {
 
+// TODO Use folly::Function for perf
+using KernelFunction = std::function<IValue(ArrayRef<IValue>)>;
```

Comment:
Err, I think neither folly::function nor std::function should be used here. These functions shouldn't be closures right? They don't have any state or closed over variables; we pass in state as an explicit argument.

Reply:
that sounds like a cool idea, thanks. Will try it out in a separate diff. Getting rid of the closure would avoid some of the perf risks here.